src/main/kotlin/anttracker/Request.kt

```diff
@@ -12,92 +12,353 @@ package anttracker.request
+    }
+    // print request to console
+    println("$affrel | $date | $name | $email | $dept")
+}
+
+// ----------------------------------------------------------------
+
+private fun displayRequestColumnTitles() {
+    // column title strings (with fixed lengths)
+    val affrel = "AffRel".padEnd(8)
+    val date = "Requested".padEnd(10)
+    val name = "Name".padEnd(30)
+    val email = "Email".padEnd(30)
+    val dept = "Dept".padEnd(10)
+
+    // print titles to console
+    println("$affrel | $date | $name | $email | $dept")
+}
+
+// ----------------------------------------------------------------
+
 // Displays a sub-menu for creating a new request and adding it to the
 //     AntTracker database.
 // Prompts the user for the various fields for the request, and validates
 //     input when necessary, re-prompting where necessary.
 // Returns the created request.
 fun enterRequestInformation(): Request? {
+    // select product for this request
     val product = selectProduct() ?: return null
+
+    // select release for this request
     val release = selectRelease(product.name) ?: return null
 
+    // select contact for this request
     var contact = selectContact()
 
     // if user doesn't select a contact, make them enter the contact information
     if (contact == null) {
         contact = enterContactInformation() ?: return null
     }
 
-    // filter available issues with product
-    val issueFilter = IssueFilter.ByProduct(product.name)
-
-    // NOTE: implement constant somewhere for "issuesPerPage"
-    var issue = selectIssue(issueFilter, 20)
+    // select issue for this request
+    var issue = selectIssue(product.name)
 
+    // check if user aborted the select issue menu: offer to create an issue instead
     if (issue == null) {
-        val issueInformation = enterIssueInformation() ?: return null
-
-        // if this returns null, then return null -- error occurred
-        issue = saveIssue(issueInformation) ?: return null
+        // if issue doesn't exist, then enter its information and create it
+        issue = enterIssueInformation(product) ?: return null
     }
 
-    var request: Request? = null
-
-    transaction {
-        request =
-            Request.new {
-                this.affectedRelease = release.id
-                this.issue = issue.id
-                this.contact = contact
-                this.requestDate = LocalDateTime.now()
-            }
+    // insert request into the db
+    val request = transaction {
+        Request.new {
+            this.affectedRelease = release.id
+            this.issue = issue.id
+            this.contact = contact
+            this.requestDate = LocalDateTime.now()
+        }
     }
 
+    // print created request to console
+    println() // blank line
     println("Created request:")
-    println("AffRel\tRequested\tName\tEmail\tDepartment")
+    displayRequestColumnTitles()
     displayRequester(request)
```

Comment:
If you already created the request above, why are you searching for it again in `displayRequester`?

Reply:
Sorry, I meant to ask why you are searching for the release version in `displayRequester`. 

The request already has the release it is associated to. You could do what is written below to extract the release id instead of having to search in the Releases table.

```kotlin
transaction { request.affectedRelease.releaseId.padEnd(8) }
```

Reply:
By the way, your definition of the affected release in the `Request` data type has to be like this.

```kotlin
var affectedRelease by Release referencedOn Requests.affectedRelease
```

This says that the `affectedRelease` in `Requests` references one of the instances of `Releases`.

Reply:
Ahh, yeah, that that would let me do what you showed above - will do this!

Reply:
done!